### ISSUE TITLE
[BUG] <issue in using rpc,but using GatewayClient is normal>

### ISSUE THREAD

**Issue by ManutdGTA** [software-mansion/starknet.py#1236]
#### the code running is GatewayClient is okay, but there is errror in using RPC node,both alchemy and infura.
```client = GatewayClient(net="mainnet") 
is correct,used 
client = FullNodeClient('https://starknet-mainnet.g.alchemy.com/v2/..')
is wrong
# 0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7 <starknet_py.net.account.account.Account object at 0x7fefe83a7f10> True
contract = await Contract.from_address(address=address, provider=provider,
                                                   proxy_config=proxy_config)
```

```
pip show starknet_py

Name: starknet-py
Version: 0.18.3
Summary: A python SDK for Starknet
Home-page: https://github.com/software-mansion/starknet.py
Author: Tomasz Rejowski
Author-email: tomasz.rejowski@swmansion.com
License: 
Location: /site-packages
Requires: aiohttp, asgiref, crypto-cpp-py, lark, marshmallow, marshmallow-dataclass, marshmallow-oneofschema, poseidon-py, pycryptodome, typing-extensions
Required-by: 
```

### Stack trace
```
'"/starknet_py/contract.py", '
 'line 613, in from_address\n'
 '    abi, cairo_version = await ContractAbiResolver(\n'
 '  File '
 '"/starknet_py/proxy/contract_abi_resolver.py", '
 'line 89, in resolve\n'
 '    return await self.resolve_abi()\n'
 '  File '
 '"/starknet_py/proxy/contract_abi_resolver.py", '
 'line 118, in resolve_abi\n'
 '    async for implementation, implementation_type in '
 'implementation_generator:\n'
 '  File '
 '"/starknet_py/proxy/contract_abi_resolver.py", '
 'line 192, in _get_implementation_from_proxy\n'
 '    raise err\n'
 '  File '
 '"/starknet_py/proxy/contract_abi_resolver.py", '
 'line 166, in _get_implementation_from_proxy\n'
 '    implementation = await proxy_check.implementation_hash(\n'
 '  File '
 '"/starknet_py/proxy/proxy_check.py", '
 'line 42, in implementation_hash\n'
 '    return await self.get_implementation(address, client)\n'
 '  File '
 '"/starknet_py/proxy/proxy_check.py", '
 'line 51, in get_implementation\n'
 '    (implementation,) = await client.call_contract(call=call)\n'
 '  File '
 '"/starknet_py/net/full_node_client.py", '
 'line 463, in call_contract\n'
 '    res = await self._client.call(\n'
 '  File '
 '"/starknet_py/net/http_client.py", '
 'line 104, in call\n'
 '    self.handle_rpc_error(result)\n'
 '  File '
 '"/starknet_py/net/http_client.py", '
 'line 111, in handle_rpc_error\n'
 '    raise ClientError(\n'
 'starknet_py.net.client_errors.ClientError: Client failed with code -32603: '
 'Internal error.\n')
```

### Steps to reproduce

```
client = GatewayClient(net="mainnet") 
is correct,used 
client = FullNodeClient('https://starknet-mainnet.g.alchemy.com/v2/..')
is wrong
# 0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7 <starknet_py.net.account.account.Account object at 0x7fefe83a7f10> True
contract = await Contract.from_address(address=address, provider=provider,
                                                   proxy_config=proxy_config)
```

### SDK Version

0.18.3

### Python version

4.10

### What operating system are you using?

Linux, Mac

### Is there an existing issue for this?

- [ ] I have searched the existing issues and verified no issue exits for this problem.

**Comment by ManutdGTA** [comment]
I am currently using my own Juno node. If I use RPC 050 http://..:6060/rpc/v0_5, there is no error. But if I use the default latest version 060, it still gives this error.

**Comment by tkumor3** [comment]
@ManutdGTA The problem is that starknet.py `0.18.3` doesn't support rpc `0.6.0`. Could you try to use starknet.py `0.19.0-alpha`? Let me know if it's working :)

**Comment by tkumor3** [comment]
@ManutdGTA Could you tell me if it resolves your problem?

**Comment by ManutdGTA** [comment]
> @ManutdGTA Could you tell me if it resolves your problem?

sry bro. I'm currently on vacation and can't write code at the moment. I'll have to ask you to handle bug testing and resolution on your own. Once you've released the official version, I'll come back to see if everything is working fine!

**Comment by ManutdGTA** [comment]
Bro I m back,there is a new error
I have updated starknet.py
used pathfinder node server latest version
```
Name: starknet-py
Version: 0.19.0

```
```
('Traceback (most recent call last):\n'
 '  File '
 '"/Users/manutdgta/code/stark_net/projects/starknet_methods.py", '
 'line 33, in async_retry\n'
 '    result = await func(**kwargs)\n'
 '  File '
 '"/Users/manutdgta/code/stark_net/projects/starknet_methods.py", '
 'line 124, in _get_contract\n'
 '    contract = await Contract.from_address(address=address, '
 'provider=provider,\n'
 '  File '
 '"/Users/manutdgta/anaconda3/envs/lib/python3.10/site-packages/starknet_py/contract.py", '
 'line 801, in from_address\n'
 '    abi, cairo_version = await ContractAbiResolver(\n'
 '  File '
 '"/Users/manutdgta/anaconda3/envs/lib/python3.10/site-packages/starknet_py/proxy/contract_abi_resolver.py", '
 'line 89, in resolve\n'
 '    return await self.resolve_abi()\n'
 '  File '
 '"/Users/manutdgta/anaconda3/envs/lib/python3.10/site-packages/starknet_py/proxy/contract_abi_resolver.py", '
 'line 118, in resolve_abi\n'
 '    async for implementation, implementation_type in '
 'implementation_generator:\n'
 '  File '
 '"/Users/manutdgta/anaconda3/envs/lib/python3.10/site-packages/starknet_py/proxy/contract_abi_resolver.py", '
 'line 192, in _get_implementation_from_proxy\n'
 '    raise err\n'
 '  File '
 '"/Users/manutdgta/anaconda3/envs/lib/python3.10/site-packages/starknet_py/proxy/contract_abi_resolver.py", '
 'line 166, in _get_implementation_from_proxy\n'
 '    implementation = await proxy_check.implementation_hash(\n'
 '  File '
 '"/Users/manutdgta/anaconda3/envs/lib/python3.10/site-packages/starknet_py/proxy/proxy_check.py", '
 'line 42, in implementation_hash\n'
 '    return await self.get_implementation(address, client)\n'
 '  File '
 '"/Users/manutdgta/anaconda3/envs/lib/python3.10/site-packages/starknet_py/proxy/proxy_check.py", '
 'line 51, in get_implementation\n'
 '    (implementation,) = await client.call_contract(call=call)\n'
 '  File '
 '"/Users/manutdgta/anaconda3/envs/lib/python3.10/site-packages/starknet_py/net/full_node_client.py", '
 'line 462, in call_contract\n'
 '    res = await self._client.call(\n'
 '  File '
 '"/Users/manutdgta/anaconda3/envs/lib/python3.10/site-packages/starknet_py/net/http_client.py", '
 'line 75, in call\n'
 '    self.handle_rpc_error(result)\n'
 '  File '
 '"/Users/manutdgta/anaconda3/envs/lib/python3.10/site-packages/starknet_py/net/http_client.py", '
 'line 82, in handle_rpc_error\n'
 '    raise ClientError(\n'
 'starknet_py.net.client_errors.ClientError: Client failed with code -32603. '
 "Message: Internal error. Data: {'error': 'Invalid message selector'}\n")
```

**Comment by tkumor3** [comment]
Referring to changes in StarkGate: ETH Token and changing contact interaction type from PROXY to ERC20. You can fetch ABI for a given contract and create a new Contract instance by calling (without proxy_config=True):
```
contract = await Contract.from_address(address= starknet_eth_address, provider=provider)
```

**Comment by ManutdGTA** [comment]
> Referring to changes in StarkGate: ETH Token and changing contact interaction type from PROXY to ERC20. You can fetch ABI for a given contract and create a new Contract instance by calling (without proxy_config=True):参考 StarkGate 中的更改：ETH 代币并将接触交互类型从 PROXY 更改为 ERC20。您可以获取给定合约的 ABI 并通过调用（不带 proxy_config=True）创建新的合约实例：
> 
> ```
> contract = await Contract.from_address(address= starknet_eth_address, provider=provider)
> ```

thx bro ,I will check this later,but how about the error `starknet_py.net.client_errors.ClientError: Client failed with code -32603. '
 "Message: Internal error. Data: {'error': 'Invalid message selector'}\n`

**Comment by tkumor3** [comment]
Could you test using alchemy and infura after adding this change with `Contract.from_address`? If problem will be steel exist, you can create the new issue
